### PR TITLE
ci: use-sem-release-nx

### PR DESCRIPTION
**Pull request by ryanbas21**
# Description
semantic-release needed multi-semantic-release which incorrectly looked
at tags for our situation which overcomplicated the process.

Switch to using the defacto nx way of handling publishing

# Type of Change

Please Delete options that are not relevant

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

# Definition of Done

Check all that apply

- [ ] Acceptance criteria is met.
- [ ] All tasks listed in the user story have been completed.
- [ ] Coded to standards.
- [ ] Code peer-reviewed.
- [ ] Ensure backward compatibility (special attention).
- [ ] API reference docs is updated.
- [ ] Unit tests are written.
- [ ] Integration tests are written.
- [ ] e2e tests are written.
- [ ] CI build passing on the feature branch.
- [ ] Functional spec is written/updated
- [ ] contains example code snippets.
- [ ] Change log updated.
- [ ] Documentation story is created and tracked.
- [ ] UI is completed or ticket is created.
- [ ] Demo to PO and team.
- [ ] Tech debts and remaining tasks are tracked in separated ticket(s).

## Documentation

- [ ] Acceptance criteria met
- [ ] Spell-check run
- [ ] Peer reviewed
- [ ] Proofread

### CODE REVIEW

File: packages/javascript-sdk/project.json [hunk 1]
```diff
@@ -55,7 +55,47 @@
           "codeCoverage": true
         }
       }
+    },
+    "version": {
+      "executor": "@jscutlery/semver:version",
+      "options": {
+        "preset": "conventional",
+        "dryRun": true,
```
Comment (ryanbas21) [inline packages/javascript-sdk/project.json:63]: Keeping as safety, we shouldn't run this anyways but i'd rather just have it as a backup

File: packages/javascript-sdk/project.json [hunk 2]
```diff
@@ -55,7 +55,47 @@
...
+          "preid": "beta",
+          "skipRootChangelog": true,
+          "push": true,
+          "postTargets": ["javascript-sdk:deploy:beta"],
+          "dryRun": true
+        }
+      }
+    },
+    "github": {
+      "executor": "@jscutlery/semver:github",
```
Comment (ryanbas21) [inline packages/javascript-sdk/project.json:80]: This pushes back to github and creates a release.

File: packages/javascript-sdk/project.json [hunk 3]
```diff
@@ -55,6 +55,48 @@
...
+        "postTargets": ["javascript-sdk:publish", "javascript-sdk:github"]
+      },
+      "configurations": {
+        "beta": {
+          "releaseAs": "prerelease",
+          "preid": "beta",
+          "skipRootChangelog": true,
+          "skipProjectChangelog": true,
+          "push": true,
+          "postTargets": ["javascript-sdk:deploy:beta"],
```
Comment (ryanbas21) [inline packages/javascript-sdk/project.json:75]: This calls deploy after the beta target is finished.

**Comment by ryanbas21** [comment]
Screen Shot 2022-05-24 at 1 00 16 PM

**Review by cerebrl — APPROVED** [review]

**Pull request merged**